minio/minio
# Fix atime support check for disk cache

- add a sleep between Stat operations to accurately detect atime

## Description
Currently atime check in disk cache is not correct due to following reasons:
1. io.Copy returns nil and not io.EOF, so atime check was passing all the time unless drive was mounted with noatime. 
2. atime check needs a sleep between the stat operations because it doesn't seem to be atomic as expected - it needs at least 1millisecond duration between stat as per testing. 
```
(base) ➜  testpgm go run atime-precision.go
Errors: 99 Success: 1 Duration: 1ns
Errors: 99 Success: 1 Duration: 1µs
Errors: 1 Success: 99 Duration: 1ms

```
## Motivation and Context
Trying to repro https://github.com/minio/minio/issues/7829 but found an issue with atime check in disk cache.  

## How to test this PR?
1. create a mount point /mnt/cache with relatime / strictatime
2. MINIO_CACHE_DRIVES=/mnt/cache ./minio server /tmp/data   should start the cache correctly
3. when mount point is changed to noatime, 2) should fail to start minio server with error `Unable to initialize disk caching: Atime support required for disk caching.`

## Types of changes

## Checklist:
